Failed to build release 0.5.0 with gcc 5.4 and default `-std` setting

Here is the diagnostic:
```
[ 50%] Building CXX object CMakeFiles/test_package.dir/test_package.cpp.o
In file included from /tmp/c3ipr/pr_66_1_0_0/test_package/test_package.cpp:1:0:
/tmp/c3ipr/pr_66_1_0_0/.conan/data/span-lite/0.5.0/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/nonstd/span.hpp: In function ‘void nonstd::span_lite::detail::throw_out_of_range(nonstd::span_lite::index_t, nonstd::span_lite::index_t)’:
/tmp/c3ipr/pr_66_1_0_0/.conan/data/span-lite/0.5.0/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/nonstd/span.hpp:614:11: error: ‘out_of_range’ is not a member of ‘std’
     throw std::out_of_range( "span::at(): index outside span" );
```

More details here: https://github.com/conan-io/conan-center-index/pull/66#issuecomment-533714103

--- a/include/nonstd/span.hpp
+++ b/include/nonstd/span.hpp
@@ -436,7 +436,7 @@ span_DISABLE_MSVC_WARNINGS( 26439 26440 26472 26473 26481 26490 )
 # include <cstdio>
 #endif
 
-#if span_CONFIG( CONTRACT_VIOLATION_THROWS_V )
+#if ! span_CONFIG( NO_EXCEPTIONS )
 # include <stdexcept>
 #endif
 

